lib/letters.ml

```diff
@@ -176,9 +176,9 @@ let ca_path_peer_verifier path =
   Lwt.return (X509.Authenticator.chain_of_trust ~time:now certs)
 ;;
 
-(* Store the detected CA cert path so it doesn't have to be detected every time a mail is
-   sent *)
-let detected_cert = ref None
+(* Store the auto-detected CA cert authenticator so it doesn't have to be detected every
+   time a mail is sent *)
+let detected_auth = ref None
```

Comment:
This is pretty much same as using global variable and should be fixed in a future release where this is encapsulated properly.

Reply:
I wrapped it into the lexical context of a closure. The new diff is horrendous, for the closure, all I did was adding the following lines to the top of `send`:
```ocaml
  (* Store the auto-detected CA cert authenticator so it doesn't have to be detected every
     time a mail is sent *)
  let detected_auth = ref None in
  fun ~config:c ~sender ~recipients ~message ->
  ```